packages/amplify-category-notifications/lib/channel-Email.js

```diff
@@ -128,11 +128,11 @@ function pull(context, pinpointApp) {
     ApplicationId: pinpointApp.Id,
   };
   spinner.start(`Pulling ${channelName} Channel.`);
-  return new Promise((resolve, reject) => {
+  return new Promise((resolve, reject) => {//eslint-disable-line
     context.exeInfo.pinpointClient.getEmailChannel(params, (err, data) => {
       if (err) {
-        spinner.fail(`get channel ${channelName} error`);
-        reject(err);
+        spinner.succeed(`no channel data retrieved for: ${channelName}`);
+        resolve(err);
       } else {
         spinner.succeed(`get ${channelName} channel successful`);
         pinpointApp[channelName] = data.EmailChannelResponse;
```

Comment:
The same code is repeated for all the channels. May be we can consolidate the logic to a single function and it take the response key as one of the params

Reply:
The API method names are different for different channels, and also the name of the property we want to grab are different in the returned data for different channels.
I'm sure there are many 'smart' ways to write this, but explicit and simple coding style makes it easier to understand and maintain by all.